docs/source/crawler/basics.rst

```diff
@@ -20,12 +20,56 @@ a modified dictionary as its output once.
 >>> next(add1_node({'input': 1}))
 {'input': 2}
 
+.. Generators::
+    Generators may have been created from various different code, including,
+    but not limited to, nodes or independent functions as long as they yield.
+    Generators may yield a dictionary, an error message, or any number of Python
+    variables. These variables may be yielded once or multiple times. A generator
+    may also return as a function does, but it must yield first.
+
 .. note::
 
    Nodes should not have side-effects, i.e. they should not modify input data,
    but yield a shallow copy if any of the field values need to be added, removed,
    or modified.  To help with creation of a new shallow copy with some fields
    adjusted, use :func:`~datalad.utils.updated`.
 
-Pipeline
---------
\ No newline at end of file
+Pipelines
+---------
+
+A pipeline is a series of generators ordered into a list. Each generator takes
+the output of its predecessor as its own input. The first function in the pipeline
+would need to be provided with specific input. The simplest pipeline could look like
+
+>>> from datalad.crawler.nodes.crawl_url import crawl_url
+>>> from datalad.crawler.nodes.matches import a_href_match
+>>> from datalad.crawler.nodes.annex import Annexificator
+
+>>> def pipeline():
```

Comment:
no need to define a function `pipeline` just yet.  Just say it is a `pipeline = [ crawl...`